File: src/entities/SeriesProductionCompany.ts
import { Entity, JoinColumn, ManyToOne } from "typeorm";
import { ProductionCompany } from "./ProductionCompany";
import { Series } from "./Series";

@Entity("series_production_company", { schema: "dionysus" })
export class SeriesProductionCompany {
  @ManyToOne(
    () => ProductionCompany,
    (productionCompany) => productionCompany.seriesProductionCompanies,
    { onDelete: "CASCADE", onUpdate: "CASCADE" }
  )
  @JoinColumn([
    {
      name: "production_company_id",
      referencedColumnName: "productionCompanyId",
    },
  ])
  productionCompany: ProductionCompany;

  @ManyToOne(() => Series, (series) => series.seriesProductionCompanies, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "series_id", referencedColumnName: "seriesId" }])
  series: Series;
}

File: engine/engine.api/Models/SeriesProductionCompany.cs
using System;
using System.Collections.Generic;

namespace engine.api.Models
{
    public partial class SeriesProductionCompany
    {
        public long? SeriesId { get; set; }
        public long ProductionCompanyId { get; set; }

        public virtual ProductionCompany ProductionCompany { get; set; } = null!;
        public virtual Series? Series { get; set; }
    }
}
